ttnn/cpp/ttnn/operations/eltwise/ternary/where.cpp

```diff
@@ -23,98 +21,57 @@ namespace ternary_utils {
 // where - ternary operator y = (predicate) ? value_true : value_false; elementwise
 // y = (predicate >= 0)*value_true + (predicate < 0)*value_false
 
-using FloatOrTensor = std::variant<Tensor, float>;
-
 Tensor where_impl(
     QueueId queue_id,
     const Tensor& predicate,
-    const FloatOrTensor& value_true,
-    const FloatOrTensor& value_false,
-    const std::optional<MemoryConfig>& output_mem_config,
-    std::optional<Tensor> output_tensor) {
-    auto get_multiplied = [&](const Tensor& condition, const FloatOrTensor& value) -> Tensor {
-        if (std::holds_alternative<Tensor>(value)) {
-            return ttnn::multiply(queue_id, condition, std::get<Tensor>(value), std::nullopt, output_mem_config);
-        } else {
-            return ttnn::multiply(queue_id, condition, std::get<float>(value), std::nullopt, output_mem_config);
-        }
+    const auto& value_true,
+    const auto& value_false,
+    const MemoryConfig& memory_config,
+    std::optional<Tensor> output) {
+    using FusedActivations = tt::stl::Span<const unary::UnaryWithParam>;
+    constexpr auto dtype = std::nullopt;
+    const auto get_multiplied = [&](const Tensor& condition, const auto& value) -> Tensor {
+        return ttnn::multiply(
```

Comment:
I wanted to be able to use in-place operations but I'm glad I tested with broadcasting because it made me realize that broadcasting rules prevented this in the general case. In the future I want to explore ways we can generically opt into in-place operations without requiring users to explicitly request it (like one possibility might be that the ttnn infra could detect when an operand has unique ownership and its properties match the requested output properties like dtype and memory config)